iree/turbine/kernel/wave/expansion.py

```diff
@@ -327,33 +327,35 @@ def _expand_reduction(
                 (reduction, get_indexed_dims(dims, expand_dims), arg_idx)
             ] = new_node
 
-            # Proceed with expansion inside the reduction
-            new_output_args.append(
-                _expand_node(
-                    arg,
-                    trace,
-                    dims,
-                    get_node_dim_scaling(arg),
-                    context,
-                    get_node_dim_scaling,
-                    res_idx,
-                )
+            expanded_output = _expand_node(
+                arg,
+                trace,
+                dims,
+                get_node_dim_scaling(arg),
+                context,
+                get_node_dim_scaling,
+                res_idx,
             )
+            # Proceed with expansion inside the reduction
+            if expanded_output in new_output_args:
```

Comment:
when do you get this situation where you expanded something that is already in the new_output_args? is it even worth expanding if it is not needed? can we use the context to detect this and prevent expanding?

Reply:
It's not getting re-expanded, but for some reason it's getting reinserted

Reply:
same as above, maybe add a TODO explaining this and we can look into this later.

Reply:
done!